src/main/java/com/hivemq/persistence/local/memory/RetainedMessageMemoryLocalPersistence.java

```diff
@@ -201,6 +201,45 @@ public void cleanUp(final int bucketIndex) {
         });
     }
 
+    @Override
+    public @NotNull BucketChunkResult<Map<String, @NotNull RetainedMessage>> getAllRetainedMessagesChunk(final int bucketIndex,
+                                                                                                         final @Nullable String ignored,
+                                                                                                         final int alsoIgnored) {
+
+        final ImmutableMap<String, RetainedMessage> collectedRetainedMessages = buckets[bucketIndex].entrySet()
```

Comment:
This does not respect the internal configuration limit, it always takes the whole bucket

Reply:
This is the same as with the other memory persistences. I added a comment with the reasoning behind.